src/NServiceBus.Core/Routing/MessageDrivenSubscriptions/MessageType.cs

```diff
@@ -96,7 +96,7 @@ public bool Equals(MessageType other)
             {
                 return true;
             }
-            return Equals(other.TypeName, TypeName) && other.Version.Major == Version.Major;
+            return Equals(other.TypeName, TypeName);
```

Comment:
food for thought: Is changing equals of a public type a breaking change?

you could argue yes

because if the type overrides equals and then changes its implementation it changes its public observed behavior when used with equals and hashcode